x/wasm/internal/keeper/genesis.go

```diff
@@ -0,0 +1,74 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmwasm/wasmd/x/wasm/internal/types"
+	// authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	// "github.com/cosmwasm/wasmd/x/wasm/internal/types"
+)
+
+// InitGenesis sets supply information for genesis.
+//
+// CONTRACT: all types of accounts must have been already initialized/created
+func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) {
+	for id, info := range data.CodeInfos {
+		bytecode := data.CodesBytes[id]
+		newId, err := keeper.Create(ctx, info.Creator, bytecode)
+		if err != nil {
+			panic(err)
+		}
+		newInfo := keeper.GetCodeInfo(ctx, newId)
+		if !info.Creator.Equals(newInfo.Creator) {
+			panic("code creators not same")
+		}
+		if bytes.Compare(info.CodeHash, newInfo.CodeHash) != 0 {
+			panic("code hashes not same")
+		}
+	}
+
+	for i, addr := range data.ContractAddresses {
+		info := data.ContractInfos[i]
+		state := data.ContractStates[i]
+
+		keeper.setContractInfo(ctx, addr, info)
+		keeper.setContractState(ctx, addr, state)
+	}
+
+}
+
+// ExportGenesis returns a GenesisState for a given context and keeper.
+func ExportGenesis(ctx sdk.Context, keeper Keeper) types.GenesisState {
+	var genState types.GenesisState
+
+	maxCodeID := keeper.GetLastID(ctx, types.KeyLastCodeID)
+	for i := uint64(0); i < maxCodeID; i++ {
```

Comment:
This should start at 1

Reply:
Is there a reason for not 0 indexing this?

Reply:
Honestly, I don't remember. I discussed this with @shanev and he implemented this, at least the start of the wasmd project (originally in cosmwasm/modules, but then I moved it over). Maybe I requested this, or maybe he did. I think it could be either way, or start at 50 for all I care. But it should be well documented, so we are consistent. Maybe we make a constant `firstCodeID`, so this is obvious?

Reply:
Hmm, the starting from `1` assumption is hard-coded into a bunch of tests.  Opened as #10 to constrain scope of this PR